ako-infra/avirest/handle_netinfo.go

```diff
@@ -491,6 +302,44 @@ func constructLsLrInCloud(lslrList []*models.Tier1LogicalRouterInfo, lslrMap map
 	return cloudLSLRList
 }
 
+func getNetworksToDeleteInCloud(objKey string, cidrToNS map[string]map[string]struct{}, client *clients.AviClient) map[string]string {
+	networks := make(map[string]string)
+	_, netModels := getAviNetFromCache(client)
+	for netName, net := range netModels {
+		pfx := lib.GetVCFNetworkName() + "-"
+		if !strings.HasPrefix(netName, pfx) {
+			continue
+		}
+		ns := strings.Split(netName, pfx)[1]
+		if _, ok := cidrToNS[ns]; !ok {
+			networks[netName] = *net.UUID
+		}
+	}
+	return networks
+}
+
+func deleteNetworkInCloud(objKey string, networksToDelete map[string]string, client *clients.AviClient) {
```

Comment:
what roles do we use for this operation? do that role have permission for this operation?

Reply:
yes, the user in WCP has admin role